From 38bd3c007622c6d349d9bb74e808ec53b43767fc Mon Sep 17 00:00:00 2001
From: Guoqing Jiang <guoqing.jiang@suse.com>
Date: Sat, 7 May 2022 15:37:07 +0800
Subject: [PATCH] Revert "virtio_net: move tx vq operation under tx queue lock"
Patch-mainline: Never, revert stable patch which is added again in sorted section
References: bsn#198
Modified-by-SEL: No


This reverts commit 4b3fd33f580fe10691b99ef0b1356b9f3607b015.

Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/virtio_net.c | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 7d1f609306f9..345a0f51e8d7 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1519,8 +1519,6 @@ static int virtnet_poll_tx(struct napi_struct *napi, int budget)
 	struct virtnet_info *vi = sq->vq->vdev->priv;
 	unsigned int index = vq2txq(sq->vq);
 	struct netdev_queue *txq;
-	int opaque;
-	bool done;
 
 	if (unlikely(is_xdp_raw_buffer_queue(vi, index))) {
 		/* We don't need to enable cb for XDP */
@@ -1530,28 +1528,10 @@ static int virtnet_poll_tx(struct napi_struct *napi, int budget)
 
 	txq = netdev_get_tx_queue(vi->dev, index);
 	__netif_tx_lock(txq, raw_smp_processor_id());
-	virtqueue_disable_cb(sq->vq);
 	free_old_xmit_skbs(sq, true);
-
-	opaque = virtqueue_enable_cb_prepare(sq->vq);
-
-	done = napi_complete_done(napi, 0);
-
-	if (!done)
-		virtqueue_disable_cb(sq->vq);
-
 	__netif_tx_unlock(txq);
 
-	if (done) {
-		if (unlikely(virtqueue_poll(sq->vq, opaque))) {
-			if (napi_schedule_prep(napi)) {
-				__netif_tx_lock(txq, raw_smp_processor_id());
-				virtqueue_disable_cb(sq->vq);
-				__netif_tx_unlock(txq);
-				__napi_schedule(napi);
-			}
-		}
-	}
+	virtqueue_napi_complete(napi, sq->vq, 0);
 
 	if (sq->vq->num_free >= 2 + MAX_SKB_FRAGS)
 		netif_tx_wake_queue(txq);
-- 
2.34.1

